File: ios/ChessClock/Theme/ChessClockFonts.swift
import SwiftUI

/// Deux familles de la planche : Instrument Serif (chronos, titres) et EB Garamond
/// (interface, libellés). Fichiers statiques embarqués dans Fonts/, licence SIL OFL,
/// déclarés dans Info.plist (UIAppFonts).
enum ChessClockFonts {
    static func instrumentSerif(_ size: CGFloat) -> Font {
        .custom("InstrumentSerif-Regular", size: size)
    }

    static func ebGaramond(_ size: CGFloat, weight: EBGaramondWeight = .regular) -> Font {
        .custom(weight.postScriptName, size: size)
    }

    enum EBGaramondWeight {
        case regular
        case medium
        case semiBold
        case italic

        var postScriptName: String {
            switch self {
            case .regular: return "EBGaramond-Regular"
            case .medium: return "EBGaramond-Medium"
            case .semiBold: return "EBGaramond-SemiBold"
            case .italic: return "EBGaramond-Italic"
            }
        }
    }
}

File: android/app/src/main/kotlin/dev/alek6dev/chessclock/ui/theme/ChessClockFonts.kt
package dev.alek6dev.chessclock.ui.theme

import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import dev.alek6dev.chessclock.R

/**
 * Deux familles de la planche : Instrument Serif (chronos, titres) et EB Garamond
 * (interface, libellés). Fichiers statiques embarqués dans res/font, licence SIL OFL.
 */
object ChessClockFonts {
    val InstrumentSerif = FontFamily(
        Font(R.font.instrument_serif_regular, FontWeight.Normal),
    )

    val EBGaramond = FontFamily(
        Font(R.font.eb_garamond_regular, FontWeight.Normal),
        Font(R.font.eb_garamond_medium, FontWeight.Medium),
        Font(R.font.eb_garamond_semibold, FontWeight.SemiBold),
        Font(R.font.eb_garamond_italic, FontWeight.Normal, FontStyle.Italic),
    )
}
